nucypher/blockchain/eth/interfaces.py

```diff
@@ -401,18 +427,38 @@ def sign_and_broadcast_transaction(self,
                                        ) -> dict:
 
         #
-        # Sign
+        # Setup
         #
 
+        # TODO # 1754
+        # TODO: Move this to singleton - I do not approve... nor does Bogdan?
+        if GlobalLoggerSettings._json_ipc:
+            emitter = JSONRPCStdoutEmitter()
+        else:
+            emitter = StdoutEmitter()
+
         if self.transacting_power is READ_ONLY_INTERFACE:
             raise self.InterfaceError(str(READ_ONLY_INTERFACE))
 
+        #
+        # Sign
+        #
+
+        if self.transacting_power.device:
+            price = unsigned_transaction['gasPrice']
+            cost_wei = price * unsigned_transaction['gas']
+            cost = Web3.fromWei(cost_wei, 'gwei')
+            # TODO: Show the USD Price
+            # Price Oracle
+            # https://api.coinmarketcap.com/v1/ticker/ethereum/
+            emitter.message(f'Confirm transaction {transaction_name} on hardware wallet... ({cost} gwei @ {price})', color='yellow')
```

Comment:
It would also be cool to show gas price information for SW accounts. Currently, this is only doing it for HW wallets.

Reply:
:heavy_check_mark: